vassal-app/src/main/java/VASSAL/counters/Decorator.java

```diff
@@ -363,6 +364,44 @@ public static GamePiece getDecorator(GamePiece p, Class<?> type) {
     }
     return null;
   }
+  
+  
+  /**
+   * @return a list of the Decorator's string/expression fields if any (for search)
+   */
+  public List<String> getExpressionList() {
+    return null; 
+  }
+
+  /**
+   * @return a list of any Message Format strings referenced in the Decorator, if any (for search)
+   */
+  public List<String> getFormattedStringList() {
+    return null;
+  }
+
+  /**
+   * @return a list of any Menu Text strings referenced in the Decorator, if any (for search)
+   */
+  public List<String> getMenuTextList() {
+    return null;
+  }
+
+  /**
+   * @return a list of any Named KeyStrokes referenced in the Decorator, if any (for search)
+   */
+  public List<NamedKeyStroke> getNamedKeyStrokeList() {
+    return null;
+  }
+
+  /**
+   * @return a list of any Property Names referenced in the Decorator, if any (for search)
+   */
+  public List<String> getPropertyList() {
+    return null;
+  }
+
```

Comment:
I wonder if you want these to return empty lists. That should make collecting all the results somewhere simpler.